rest/route/host_agent.go

```diff
@@ -525,6 +525,26 @@ func assignNextAvailableTask(ctx context.Context, env evergreen.Environment, tas
 				"host_id":   currentHost.Id,
 			}))
 
+			if nextTask.IsStuckTask() {
+				if err := model.EndAndResetSystemFailedTask(ctx, env.Settings(), nextTask, evergreen.TaskDescriptionStranded); err != nil {
```

Comment:
I think what we want here is `ClearAndResetStrandedHostTask` rather than `EndAndResetSystemFailedTask` because `EndAndResetSystemFailedTask` doesn't de-assign the task from the host.